### DISCUSSION TITLE
Can(CanFd) for firmware update

### DISCUSSION THREAD

**Discussion by CEmir10** [zephyrproject-rtos/zephyr#47773] [Q&A] [answered]
Is there someone that works on Can protocol for firmware update?

**Comment by str4t0m** [comment, marked as answer]
If you are using CANOpen, there is support for the program download specified in CiA 302-3.
I am not aware of any other efforts at the moment.

https://docs.zephyrproject.org/latest/samples/modules/canopennode/README.html#testing-canopen-program-download

**Reply by henrikbrixandersen** [reply]
CANopen program download (CiA 302-3) is supported by Zephyr via CANopenNode: https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/canopennode/canopen_program.c
